vault/resource_database_secret_backend_connection.go

```diff
@@ -561,6 +566,57 @@ func getDatabaseSchema(typ schema.ValueType) schemaMap {
 			MaxItems:      1,
 			ConflictsWith: util.CalculateConflictsWith(dbEngineOracle.Name(), dbEngineTypes),
 		},
+		dbEngineRedis.name: {
+			Type:        typ,
+			Optional:    true,
+			Description: "Connection parameters for the redis-database-plugin plugin.",
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"host": {
+						Type:        schema.TypeString,
+						Required:    true,
+						Description: "Specifies the host to connect to",
+					},
+					"port": {
+						Type:         schema.TypeInt,
+						Optional:     true,
+						Description:  "The transport port to use to connect to Redis.",
+						Default:      6379,
+						ValidateFunc: validation.IsPortNumber,
+					},
+					"username": {
+						Type:        schema.TypeString,
+						Required:    true,
+						Description: "Specifies the username for Vault to use.",
+					},
+					"password": {
+						Type:        schema.TypeString,
+						Required:    true,
+						Description: "Specifies the password corresponding to the given username.",
+						Sensitive:   true,
+					},
+					"tls": {
+						Type:        schema.TypeBool,
+						Optional:    true,
+						Description: "Specifies whether to use TLS when connecting to Redis.",
+						Default:     false,
+					},
+					"insecure_tls": {
+						Type:        schema.TypeBool,
+						Optional:    true,
+						Description: "Specifies whether to skip verification of the server certificate when using TLS.",
+						Default:     false,
+					},
+					"ca_cert": {
```

Comment:
We have a few unrelated PRs out where the user would like to be able to pass this in as a string, wonder if it would be worth supporting both forms? Reading a file from a Terraform file is not always supported in a pipeline, and requiring it can make the deployment a little more cumbersome.

Reply:
I will change the description. This is actually doing what you are referring to. We are not passing the path to the actual file, but the contents. I will update the description to say that. Thank you for the suggestion 👍